actors/builtin/miner/miner_actor.go

```diff
@@ -142,66 +140,37 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 }
 
 //////////////////
-// SurprisePoSt //
+// FallbackPoSt //
 //////////////////
 
-// Called by Actor to notify StorageMiner of SurprisePoSt Challenge.
-func (a Actor) OnSurprisePoStChallenge(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
-	rt.ValidateImmediateCallerIs(builtin.StoragePowerActorAddr)
-
-	var st State
-	challenged := rt.State().Transaction(&st, func() interface{} {
-		// If already challenged, do not challenge again.
-		// Failed PoSt will automatically reset the state to not-challenged.
-		if st.PoStState.isChallenged() {
-			return false
-		}
-
-		st.PoStState = PoStState{
-			LastSuccessfulPoSt:     st.PoStState.LastSuccessfulPoSt,
-			SurpriseChallengeEpoch: rt.CurrEpoch(),
-			NumConsecutiveFailures: st.PoStState.NumConsecutiveFailures,
-		}
-		return true
-	}).(bool)
-
-	if challenged {
-		// Request deferred Cron check for SurprisePoSt challenge expiry.
-		cronPayload := CronEventPayload{
-			EventType: CronEventType_Miner_SurpriseExpiration,
-			Sectors:   nil,
-		}
-		surpriseDuration := power.SurprisePostChallengeDuration
-		a.enrollCronEvent(rt, rt.CurrEpoch()+surpriseDuration, &cronPayload)
-	}
-	return &adt.EmptyValue{}
-}
-
-type SubmitSurprisePoStResponseParams struct {
-	onChainInfo abi.OnChainSurprisePoStVerifyInfo
-}
+// TODO: where should this live?
+const ProvingPeriod = 300
 
-// Invoked by miner's worker address to submit a response to a pending SurprisePoSt challenge.
-func (a Actor) SubmitSurprisePoStResponse(rt Runtime, params *SubmitSurprisePoStResponseParams) *adt.EmptyValue {
+// Invoked by miner's worker address to submit their fallback post
+func (a Actor) SubmitFallbackPoSt(rt Runtime, params *abi.OnChainPoStVerifyInfo) *adt.EmptyValue {
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
 		rt.ValidateImmediateCallerIs(st.Info.Worker)
-		if !st.PoStState.isChallenged() {
-			rt.Abort(exitcode.ErrIllegalState, "Not currently challenged")
+
+		if rt.CurrEpoch() <= st.PoStState.ProvingPeriodStart+ProvingPeriod {
```

Comment:
@whyrusleeping note that we have 

```
// The average period (i.e. 1/frequency) of surprise PoSt challenges to each miner.
const SurprisePoStPeriod = abi.ChainEpoch(5760) // ~2 days @ 30 second epochs. PARAM_FINISH

// The time a miner has to respond to a surprise PoSt challenge.
const SurprisePostChallengeDuration = abi.ChainEpoch(240) // ~2 hours @ 30 second epochs. PARAM_FINISH

// The minimum period after a PoSt challenge before a miner can be challenged again.
const SurprisePoStNoChallengePeriod = abi.ChainEpoch(240) // PARAM_FINISH
```

the `NoChallengePeriod` should be removed, but the `period` and `challengeDuration` can be adapted to our ends here. To @anorth 's point, expliciting the duration rather than doing so implicitly by offsetting starts off one another would be v helpful.

My assumption as to how this should work is:

```
| PP1    > CP1 | PP2     > CP2| ...
```
where PP is proving period, and CP is challenge period, so if PP = 500, and CP = 10, you would get challenged at epoch 490 of the PP... Feel free to correct my understanding

Reply:
I see below that you have the CP part of the next PP, why.... Either works. Seems more confusing :b (means we have to say "you will be challenged once per PP", rather than "you will post once per PP")

Reply:
Take another look at this code, I went through it with @zixuanzh today